src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -1313,11 +1312,30 @@ private static DateTime ParseTimeOfDay(ReadOnlySpan<char> time)
                     // 0                30 31              58 59              89      334            364
                     // |-------Jan--------|-------Feb--------|-------Mar--------|....|-------Dec--------|
                     //
-                    //
                     // For example if n is specified as 60, this means in leap year the rule will start at Mar 1,
                     // while in non leap year the rule will start at Mar 2.
                     //
-                    // If we need to support n format, we'll have to have a floating adjustment rule support this case.
+                    // This n Julian day format is very uncommon and mostly  used for convenience to specify dates like January 1st
+                    // which we can support without any major modification to the Adjustment rules. We'll support this rule  for day
+                    // numbers less than 59 (up to Feb 28). Otherwise we'll skip this POSIX rule.
+                    // We've never encountered any time zone file using this format for days beyond Feb 28.
+
+                    if (int.TryParse(date, out int julianDay) && julianDay < 59)
```

Comment:
Per the tzset man page pasted above, the value seems to start at 1, not 0..?

I may just be confused though, as I don't think I understand why the J format exists, unless it's a way to express a day that ignores a leap day (which is the way I read the man page). Eg., your example `Africa\Casablanca: <+00>0<+01>,0/0,J365/25` means from the first day of the year to the last day of the year, whether it's a leap year or not. Also, this example would not be handled by this code as 365 is larger than 58 anyway?

cc @tarekgh

Reply:
I believe the Jn format was already supported, this section is adding the `n` format. Which based on the man page is:

```
  n      This specifies the zero-based Julian day with n between 0 and 365.  February 29 is counted in leap years.
```

Which is a regular int number. The Jn format which you are referring to is handled after this if: https://github.com/dotnet/runtime/blob/8e690a261f4f167a9829a60c0e4989a64498fbf7/src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs#L1502

Which is when `date[0] == 'J'`.

Reply:
What about the question about only supporting up to Feb 28, when the example given is 365 ? I may well be still confused haha

Reply:
In this bizarre counting system, Feb 29 doesn't exist.  Dec 31st is always day 365, even in a leap year.

Reply:
@danmoseley we are full support Jn rule and partially support n rule. Jn rule is not differentiating between leap year or non-leap years. So, Feb 29th cannot express with the Jn rule. i.e. if using J60 it is always March 1st and never be Feb 29th.  

In the other hand, n rule account for leap years. note as mentioned before this rule is zero based rule. if specify `59` it means March 1st when the year is not leap year and will mean Feb 29th in the leap years. anyway, I never encountered any time zone file use this rule to specify a day after Feb 28th. 

Let me know if there is anything unclear here.